### ISSUE TITLE
[Feature Request] Passing falsy `keySeparator` to i18next.t to disable the keySeparator option in specific call

### ISSUE THREAD

**Issue by KagamiChan** [i18next/i18next#1021]
reproduction code with i18next 10.4.1:
```js
i18next.init({
  lng: 'en',
  defaultNS: 'main',
  resources: {
    'en': {
      main: {
        foo: {
          foo: 'bar',
        },
        'bar.baz': 'hello',
      },
    },
  },
}, () => {
  console.log(i18next.t('foo.foo')) // bar
  console.log(i18next.t('bar.baz', { keySeparator: false })) // bar.baz
  console.log(i18next.t('bar.baz', { keySeparator: true }))  // hello
})
```

In our projects we might have dots in keys for some namespaces / components, although escaping both resources and keys is a possible solution, passing `{ keySeparator: false}` to specific i18next.t usage seems more natural

**Comment by jamuhl** [comment]
keySeparator needs to be set to some char when passed via t function -> eg. if not like to have it set it to something not used:

`console.log(i18next.t('bar.baz', { keySeparator: '###' })) // hello`

that it does not disable it when setting to false is caused by https://github.com/i18next/i18next/blob/master/src/Translator.js#L57 placed around all places -> you might create a PR changing that to support setting it to false (setting to true won't do anything meaningful as it is expected to be a string)

**Comment by KagamiChan** [comment]
@jamuhl Thanks for your reply. Will take a look at the code

**Comment by jamuhl** [comment]
https://github.com/i18next/i18next/pull/1051

published in 11.3.1
